tests/ui/shell-argfiles/shell-argfiles-badquotes.rs

```diff
@@ -1,6 +1,6 @@
 // Check to see if we can get parameters from an @argsfile file
 //
-// compile-flags: --cfg cmdline_set -Z shell-argfiles @shell:{{src-base}}/shell-argfiles/shell-argfiles-badquotes.args
+// compile-flags: --cfg cmdline_set -Z shell-argfiles @shell:{{src-base}}{{sep}}shell-argfiles/shell-argfiles-badquotes.args
```

Comment:
Wait, why is `{{sep}}` here and also `/`?

Reply:
I really do not understand how path sanitization is working here lol, or frankly why we're able to pass a forward-slashed path on windows at all 😵

Reply:
It's very unfortunate. When post-processing stderr output, compiletest is replacing `test_file.parent()` with `"$DIR"`. Because we're in a subdirectory of `src-base` aka `tests/ui`, our parent directory on windows is `src\tests\ui\shell-argfiles`. But the path our diagnostic prints is literally what the user specified: `src\tests\ui/shell-argfiles/shell-argfiles-badquotes.stderr`. Note that here, there's a forward slash between `ui` and the test directory `shell-argfiles`. So when compiletest does the replacement, it doesn't replace the path in the emitted error since the slash directions differ. The "easy" way to fix this is to use the native path separator between `{{src-base}}` and our test directory. That will make sure that compiletest's `$DIR` substitution gets a match and replaces it.

Windows unfortunately doesn't really care if you use forward slashes or backslashes for its path. So this issue has flown under the radar for a while. We can still open files with `/` as a path separator and do everything normally, it's just the stderr output that doesn't get replaced with `$DIR`.

This seemed like the least-intrusive way to work around this issue for the time being. I'd be happy to do a follow-up with a more general fix for Windows if that's desirable.

Reply:
Wait, can you turn this into two test revisions `//[windows]` and `//[not-windows]` perhaps?

Reply:
Ooh I like that idea. I'll whip that up quick and leave a comment explaining the situation.